src/shogun/lib/SGVector.cpp

```diff
@@ -210,8 +202,7 @@ COMPLEX128_ERROR_TWOARGS(random)
 template <class T>
 index_t SGVector<T>::find_position_to_insert(T element)
 {
-	if (on_gpu())
-		SG_SERROR("Find_position_to_insert method not available for vectors on GPU.\n")
+	on_gpu_check();
```

Comment:
@vigsterkr you commented that you might allow for those guys.
I kind of think this is dangerous, and would like to enforce to use an explicit transfer back before allowing to access the data. This way, we make sure people are aware that this is expensive.
That said, all methods of SGVector should go away in my eyes, and be moved to linalg. `SGVector` should just be memory wrapping and interfaces, nothing more, in particular no numerics or linear algebra.
But we can do that in a later patch @OXPHOS